### PR TITLE
Refactored the Mounter::ApplicationWrapper

### PR DESCRIPTION
**Pull request by hcatlin**
I didn't go as far as I could, but I basically had the ApplicationWrapper (and therefore Padrino) register itself with the application that is being mounted by setting a ```@padrino_wrapper``` class variable on the application. This means that you can customize your app to work better within a Padrino environment.

### CODE REVIEW

File: padrino-core/lib/padrino-core/mounter.rb [hunk 1]
```diff
@@ -39,25 +43,35 @@ def app_file
...
+
+      def require_dependencies
+        Padrino.require_dependencies(dependencies, :force => true)
       end
 
       def setup_application!
         @configured ||=
           begin
             $LOAD_PATH.concat(prerequisites)
+<<<<<<< HEAD
```
Comment (ujifgc) [inline padrino-core/lib/padrino-core/mounter.rb]: Right…

Comment (hcatlin) [inline padrino-core/lib/padrino-core/mounter.rb]: Yeah... I saw this just as I had committed. :(

File: padrino-core/lib/padrino-core/mounter.rb [hunk 2]
```diff
@@ -39,25 +43,31 @@ def app_file
 
       def root
         return @__root if @__root
-        obj = __getobj__
         @__root = obj.respond_to?(:root) ? obj.root : File.expand_path("#{app_file}/../")
       end
 
+      def obj
```
Comment (ujifgc) [inline padrino-core/lib/padrino-core/mounter.rb]: Now, why do we need this method exactly?

**Comment by ujifgc** [comment]
We will probably won't be able to merge this PR after a5b260fe6fab7b96f91d1af85534fd660ee4a7db. Thank you for addressing the issue though.

**Pull request closed without merging**